tests/modules/nanopb/lib/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Already done in the application folder but should be safe from any directory
+list(APPEND CMAKE_MODULE_PATH ${ZEPHYR_BASE}/modules/nanopb)
+include(nanopb)
+
+zephyr_library_named(mylib)
+zephyr_library_sources(lib.c)
+
+# Demonstrate that our library includes a generated header so we need the following dependency
+add_dependencies(mylib nanopb_generated_headers)
+
+# Add include directory to find the generated headers
+zephyr_library_include_directories(${CMAKE_BINARY_DIR})
```

Comment:
@tejlmand is it allowed/recommended to use `zephyr_library_` cmake functions in libraries nested in the application or should we only use vanilla cmake functions:

```cmake
add_library(mylib lib.c)

# Demonstrate that our library includes a generated header so we need the following dependency
add_dependencies(mylib nanopb_generated_headers)

# Add include directory to find the generated headers
target_include_directories(mylib PRIVATE ${CMAKE_BINARY_DIR})

# Link against zephyr
target_link_libraries(mylib zephyr)

```

Reply:
> @tejlmand is it allowed/recommended to use zephyr_library_ cmake functions in libraries nested in the application or should we only use vanilla cmake functions:

Generally I would not recommend it.
Especially not the `zephyr_library()` function, as calling that after `find_package()` is not behaving as most users thinks.

The `zephyr_library()` are usually linked inside `whole-archive` but not when called from the app's CMakeLists.txt tree.

Reply:
I will modify using the vanilla cmake functions.